src/librustc/mir/visit.rs

```diff
@@ -797,8 +797,9 @@ macro_rules! make_mir_visitor {
             ) {
                 let UserTypeProjection {
                     ref $($mutability)* base,
+                    projs: _, // Note: Does not visit projection elems!
```

Comment:
Is there any particular reason for this? I guess just because nobody has any reason to.

Reply:
Well, the existing MIR visit methods take a `PlaceElem`, not a `ProjectionElem<(), ()>`. So it would have been a bit of extra complexity in the visit API to try to support it ... and to what end?  There isn't any recursive structure to visit, and you would probably want to know all about the surrounding context when inspecting any particular `ProejctionElem<(), ()>` within the `UserTypeProjection` ... so better to just do it explicitly in one's `visit_user_type_projection` method?